ni.zsh

```diff
@@ -571,18 +571,88 @@ function ni-dlx(){
 
 # auto completion
 function _ni(){
-  # ni <subcommands>
-  local -a subcommands
-  subcommands=(
-    'add:add package'
-    'run:run scripts'
-    'test:run test script'
-    'upgrade:upgrade package'
-    'upgrade-interactive:upgrade package interactively'
-    'remove:remove package'
-    'exec:execute command'
-    'dlx:download package and execute command'
-  )
-  _describe -t subcommands 'subcommands' subcommands
+  local context state state_descr line
+  typeset -A opt_args
+
+  _arguments -C \
+    '1: :->cmds' \
+    '*:: :->args'
+
+  case "$state" in
+    cmds)
+      # ni <subcommands>
+      local -a subcommands
+      subcommands=(
+        'add:add package'
+        'run:run scripts'
+        'test:run test script'
+        'upgrade:upgrade package'
+        'upgrade-interactive:upgrade package interactively'
+        'remove:remove package'
+        'exec:execute command'
+        'dlx:download package and execute command'
+      )
+      _describe -t subcommands 'subcommands' subcommands
+      ;;
+    args)
+      case $line[1] in
+        add)
+          # ni add <package>
+          _arguments \
+            '--dev[Install as development dependency]' \
+            '-D[Install as development dependency]'
+          ;;
+        remove|upgrade)
+          # ni remove <package>
+          # ni upgrade <package>
+          if [ -f "package.json" ]; then
+            if command -v jq >/dev/null 2>&1; then
```

Comment:
`jq` is requirements of ni.zsh.
https://github.com/azu/ni.zsh?tab=readme-ov-file#requirements

If there is no jq, it is ok to fast fall.

```
If jq is not 
  echo “jq is required!
  exit?
```

Reply:
Fixed: [4cdf0f3](https://github.com/azu/ni.zsh/pull/22/commits/4cdf0f326b8f243291f73d6eb8f8f8f1d318c809)

Reply:
Can we move `if command -v jq >/dev/null 2>&1; then` to above `case` statement?

```zsh
function _ni(){
  if command -v jq >/dev/null 2>&1; then
    echo "jq command is required to use ni.zsh"
    return 1
  fi
...

```

Is it possible to apply early return pattern?

Reply:
I think there's no problem.  
I have modified the code accordingly.